### ISSUE TITLE
Does Fairseq support gradient checkpointing?

### ISSUE THREAD

**Issue by PersianNLPer** [pytorch/fairseq#1124]
Hello,
As far as I understood by reading the Pytorch docs and some other material, the gradient checkpointing seems to be the way to handle very large input strings. Moreover, it seems that Pytorch already supports it natively so no need to do big hacks in order to implement it.

So, may I ask if it is already implemented in Fairseq?

Thank you very much,
Amin

**Comment by myleott** [comment]
Yes, checkpointing is done entirely in the model, so you just need to follow the pytorch docs and update the model code appropriately. We have done this for for some recent projects, but the checkpointing code can get a bit messy, so we haven't merged it to master.
